### PR TITLE
Add email validation

### PR DESCRIPTION
**Pull request by arvryna**
* Handle email validation
* Also handle in case if the user tries to provide email address with a domain name
* Fixes https://github.com/requiemofthesouls/pigeomail/issues/4

### CODE REVIEW

**Comment by arvryna** [comment]
image

File: internal/telegram/create.go [hunk 1]
```diff
@@ -49,17 +52,32 @@ func (b *Bot) handleCreateCommandStep1(update *tgbotapi.Update) {
...
-func validateMailboxName() bool {
-	// TODO: email validation
-	return true
+func validateMailboxName(email string) bool {
+	// Adding a simple stub string, as validation requires full email address
+	// since user will give us just the name of inbox
+	domainStub := "@pigeomail.com"
+	_, err := mail.ParseAddress(email + domainStub)
+	return err == nil
 }
```
Comment (requiemofthesouls) [inline internal/telegram/create.go:61]:
```suggestion
func (b *Bot) validateMailboxName(email string) bool, string {
	if strings.Contains(email, "@") {
		return false, "please enter mailbox name without domain"
	}
	
	if _, err := mail.ParseAddress(email + "@" + b.domain); err != nil {
	    return false, email + " is not a valid name for mailbox, please choose a new one"
	}
	
	return true
}
```

File: internal/telegram/create.go [hunk 2]
```diff
@@ -49,17 +52,32 @@ func (b *Bot) handleCreateCommandStep1(update *tgbotapi.Update) {
...
+		b.api.Send(msg)
+		return
+	}
+
+	if !validateMailboxName(update.Message.Text) {
+		msg.Text = fmt.Sprintf("<%s> is not a valid name for email inbox, please choose a new one.", update.Message.Text)
+		b.api.Send(msg)
+		return
+	}
+
```
Comment (requiemofthesouls) [inline internal/telegram/create.go:77]:
```suggestion
	if ok, text := validateMailboxName(update.Message.Text); !ok {
		msg.Text = text
		b.api.Send(msg)
		return
	}
```

**Comment by arvryna** [comment]
image

**Review by requiemofthesouls — APPROVED** [review]
nice

**Pull request merged**